newrelic-alerts-configurator-dsl/src/main/kotlin/com/ocadotechnology/newrelic/alertsconfigurator/dsl/configuration/condition/NrqlConditionConfigurationDsl.kt

```diff
@@ -22,10 +23,14 @@ fun nrqlSignalConfiguration(block: NrqlSignalConfigurationDsl.() -> Unit): NrqlS
     val dsl = NrqlSignalConfigurationDsl()
     dsl.block()
 
-    return NrqlSignalConfiguration.builder()
-        .aggregationWindow(requireNotNull(dsl.aggregationWindow) { "Aggregation window cannot be null" })
-        .evaluationWindows(requireNotNull(dsl.evaluationWindows) { "Evaluation window cannot be null" })
-        .signalFillOption(requireNotNull(dsl.signalFillOption) { "Signal fill option cannot be null" })
-        .signalLostConfiguration(requireNotNull(dsl.signalLostConfiguration) { "Signal lost configuration cannot be null" })
+    val signalConfigurationBuilder = NrqlSignalConfiguration.builder()
+            .aggregationWindow(requireNotNull(dsl.aggregationWindow) { "Aggregation window cannot be null" })
+            .evaluationWindows(requireNotNull(dsl.evaluationWindows) { "Evaluation window cannot be null" })
+            .signalFillOption(requireNotNull(dsl.signalFillOption) { "Signal fill option cannot be null" })
+            .signalLostConfiguration(requireNotNull(dsl.signalLostConfiguration) { "Signal lost configuration cannot be null" })
+    if (dsl.signalFillOption == SignalFillOption.STATIC) {
+        signalConfigurationBuilder.signalFillValue(requireNotNull(dsl.signalFillValue))
```

Comment:
To be constistent, can you add some error message to infrom that signallFillVaslue is required with STATIC fill options?

Reply:
Sure